### PR TITLE
CircleCI 2.0 compatibility

### PR DESCRIPTION
**Pull request by heavelock**
### What does this PR do?
Changes the config of CircleCI to be compatible with CircleCI 2.0. The CircleCI 1.0 was EOLed on 31st of Agust, 2018.
https://circleci.com/blog/sunsetting-1-0/

### Why was it initiated?  Any relevant Issues?
Nope.
### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+ DOCS"
- [x] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [x] All tests still pass.
- [x] Any new features or fixed regressions are be covered via new tests.
- [x] Any new or changed features have are fully documented.
- [x] Significant changes have been added to `CHANGELOG.txt` .
- [x] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by heavelock** [comment]
Okay, ready for review.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+
+      - run:
+          name: install dependencies
+          command: |
+            conda create -n test-env
+            source activate test-env
```
[the remark this answers is not in the document]
Comment (heavelock) [reply .circleci/config.yml]: Oh, I was not even aware that they unified linux with windows in terms of that. Thanks!

**Comment by heavelock** [comment]
Well, it seems that `conda activate` doesn't work on this docker image. The conda itself proposes adding lines to `.bashrc` but it will be easier to revert to `source activate` instead.

**Comment by megies** [comment]
> Well, it seems that `conda activate` doesn't work on this docker image. The conda itself proposes adding lines to `.bashrc` but it will be easier to revert to `source activate` instead.

Ah OK, yeah, then just use the old style, I guess. Whatever works..

**Comment by krischer** [comment]
Cool! Thanks for this :-)

**Pull request merged**